### ISSUE TITLE
Different uses of `timeout` on exporters

### ISSUE THREAD

**Issue by mx-psi** [open-telemetry/opentelemetry-collector#4497]
The `exporterhelper` includes a timeout mechanism, which is enabled by default (see #4455 for more on that), and sets a timeout for the whole `Consume[Metrics/Traces/Logs]` function call. Several exporters in contrib however rely on Go's HTTP client timeout instead ([1][1], [2][2], [3][3], [4][4], [5][5]), while still using the same `timeout` option.

This is an issue to discuss:
1. If a different option name should be used for these exporters use case,
2. if we decide that using the same name is okay, should these exporters avoid using `exporterhelper.TimeoutSettings` and use a custom timeout field instead?

This was originally brought up by @codeboten at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6414#discussion_r759736023. For context, in our exporter we don't rely on the global timeout mechanism because we need to do several network calls on a single `Consume` call and want to timeout on each specific call instead.

[1]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/82142b9f8b991866c9467e719d69922435dc1fcc/exporter/datadogexporter/factory.go#L157-L158
[2]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/243d6423992a4a76db30eb254d715ccb3a23c9e4/exporter/observiqexporter/exporter.go#L42-L43
[3]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/243d6423992a4a76db30eb254d715ccb3a23c9e4/exporter/splunkhecexporter/factory.go#L105-L106
[4]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/243d6423992a4a76db30eb254d715ccb3a23c9e4/exporter/lokiexporter/factory.go#L71-L72
[5]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/243d6423992a4a76db30eb254d715ccb3a23c9e4/exporter/signalfxexporter/factory.go#L121-L122

**Comment by jpkrohling** [comment]
@MovieStoreGuy, this would be a good entry to add to the doc you are creating.

**Comment by MovieStoreGuy** [comment]
Yeah, I remember seeing this in the kafka exporter and thought it was strange.

I do think I like the idea of option 1 though, having `OperationTimeout` and `NetworkRequestTimeout` would help clarify things as I do think each has a purpose and conflating the two would almost require User's to read and understand the setup code.

**Comment by jpkrohling** [comment]
@codeboten, are you able to summarize what was discussed yesterday? I had Zoom problems and missed part of the discussion.

**Comment by MovieStoreGuy** [comment]
Hey @codeboten , 


Was there more to add to this please?
